File: src/services/FriendService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/';

const getFriends = (userId) => {
    return axios.get(`${API_URL}users/${userId}/friends`);
};

const addFriend = (userId, friendId) => {
    return axios.post(`${API_URL}users/${userId}/friends/${friendId}`);
};

const FriendService = {
    getFriends,
    addFriend
};

export default FriendService;

File: src/main/java/com/example/CrissX/service/FriendService.java
package com.example.CrissX.service;

import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.example.CrissX.model.Friend;
import com.example.CrissX.model.User;
import com.example.CrissX.repository.FriendRepository;
import com.example.CrissX.repository.UserRepository;

@Service
public class FriendService {

    @Autowired
    private FriendRepository friendRepository;

    @Autowired
    private UserRepository userRepository;

    public Friend addFriend(Long userId, Long friendId) {
        User user = userRepository.findById(userId).orElseThrow(() -> new RuntimeException("User not found"));
        User friend = userRepository.findById(friendId).orElseThrow(() -> new RuntimeException("Friend not found"));
        
        Friend friendEntity = new Friend();
        friendEntity.setUser(user);
        friendEntity.setFriend(friend);
        
        return friendRepository.save(friendEntity);
    }

    public List<Friend> getFriends(Long userId) {
        return friendRepository.findAll();
    }

    // Additional friend management methods (remove friend, etc.)
}
